muffnn/mlp/base.py

```diff
@@ -246,14 +265,20 @@ def _set_up_graph(self):
                 if self.keep_prob != 1.0:
                     t = tf.nn.dropout(t, keep_prob=self._keep_prob)
                 t = affine(t, layer_sz, scope='layer_%d' % i)
+                self._checks.append(tf.check_numerics(t, 'layer_%d' % i))
             t = t if self.activation is None else self.activation(t)
 
         # The output layer and objective function depend on the model
         # (e.g., classification vs regression).
         t = self._init_model_output(t)
         self._init_model_objective_fn(t)
 
-        self._train_step = tf.train.AdamOptimizer().minimize(self._obj_func)
+        self._train_step = tf.train.AdamOptimizer()
+        self._grads_and_vars = self._train_step.compute_gradients(self._obj_func, tf.get_collection(tf.GraphKeys.TRAINABLE_VARIABLES))
+        # for a, b in self._grads_and_vars:
+        #     self._checks.append(tf.check_numerics(a, a.name))
+        #     self._checks.append(tf.check_numerics(b, b.name))
+        self._train_step_step = self._train_step.apply_gradients(self._grads_and_vars)
```

Comment:
We only split the train step up for debugging. You should put all of this back to the original form.